sklearn/discriminant_analysis.py

```diff
@@ -242,18 +242,27 @@ class LinearDiscriminantAnalysis(BaseEstimator, LinearClassifierMixin,
     >>> clf = LinearDiscriminantAnalysis()
     >>> clf.fit(X, y)
     LinearDiscriminantAnalysis(n_components=None, priors=None, shrinkage=None,
-                  solver='svd', store_covariance=False, tol=0.0001)
+                  solver='svd', store_covariances=False, tol=0.0001)
     >>> print(clf.predict([[-0.8, -1]]))
     [1]
     """
+
+    import warnings
+
     def __init__(self, solver='svd', shrinkage=None, priors=None,
-                 n_components=None, store_covariance=False, tol=1e-4):
+                 n_components=None, store_covariances=False,
+                 store_covariance=None, tol=1e-4):
         self.solver = solver
         self.shrinkage = shrinkage
         self.priors = priors
         self.n_components = n_components
-        self.store_covariance = store_covariance  # used only in svd solver
+        self.store_covariances = store_covariances  # used only in svd solver
         self.tol = tol  # used only in svd solver
+        if store_covariance is not None:
```

Comment:
It's also possible that we need to handle the case where this is set, but not in the constructor (e.g. as an attribute or with `set_params`). I think we should leave this warning until `fit`.